src/jmespath.net/Functions/JmesPathFunctionFactory.cs

```diff
@@ -61,17 +62,24 @@ public IRegisterFunctions Register(string name, JmesPathFunction function)
 
         public IRegisterFunctions Register<T>() where T : JmesPathFunction
         {
-            var instance = Activator.CreateInstance<T>();
-            Register(instance.Name,instance);
+            var ctor = typeof(T).GetConstructor(new Type[] { typeof(IScopeParticipant), });
+            var instance = (T)(
+                (ctor != null)

```

Comment:
If the function has a `ctor` accepting an instance of `IScopeParticipant`, inject the corresponding instance.
This keeps compatibility with functions that are not aware of scopes, even third-party ones.